lint/src/test/kotlin/io/homeassistant/lint/serialization/MissingSerializableAnnotationIssueTest.kt

```diff
@@ -0,0 +1,329 @@
+1 errors, 0 warnings""",
+            )
+    }
+
+    @Test
+    fun `Given a class without Serializable annotation within a Map when invoking kotlinJsonMapper encodeToString then MissingSerializableAnnotationIssue is raised`() {
+        lint().issues(*issuesToUse)
+            .allowMissingSdk()
+            .files(
+                serializableStub,
+                kotlinJsonMapperStub,
+                jsonStub,
+                kotlin(
+                    """
+                package io.homeassistant
+                import io.homeassistant.companion.android.common.util.kotlinJsonMapper
+                import kotlinx.serialization.json.SerializationStrategy
+                
+                class Home
+                fun main() {
+                    val value = kotlinJsonMapper.encodeToString(mapOf<String, Home>( "hello" to Home()))
+                    println(value)
+                }
+                """,
+                ).indented(),
+            )
+            .run()
+            .expect(
+                """src/io/homeassistant/Home.kt:7: Error: The class io.homeassistant.Home is missing the @Serializable annotation. [MissingSerializableAnnotation]
+    val value = kotlinJsonMapper.encodeToString(mapOf<String, Home>( "hello" to Home()))
+                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+1 errors, 0 warnings""",
+            )
+    }
+
+    @Test
+    fun `Given a list of a class without Serializable annotation within a Map when invoking kotlinJsonMapper encodeToString then MissingSerializableAnnotationIssue is raised`() {
+        lint().issues(*issuesToUse)
+            .allowMissingSdk()
+            .files(
+                serializableStub,
+                kotlinJsonMapperStub,
+                jsonStub,
+                kotlin(
+                    """
+                package io.homeassistant
+                import io.homeassistant.companion.android.common.util.kotlinJsonMapper
+                import kotlinx.serialization.json.SerializationStrategy
+                import java.util.List
+                
+                class Home
+                fun main() {
+                    val value = kotlinJsonMapper.encodeToString(mapOf<String, List<Home>>( "hello" to Home()))
+                    println(value)
+                }
+                """,
+                ).indented(),
+            )
+            .run()
+            .expect(
+                """src/io/homeassistant/Home.kt:8: Error: The class io.homeassistant.Home is missing the @Serializable annotation. [MissingSerializableAnnotation]
+    val value = kotlinJsonMapper.encodeToString(mapOf<String, List<Home>>( "hello" to Home()))
+                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+1 errors, 0 warnings""",
+            )
+    }
+
+    @Test
+    fun `Given the usage of any when invoking kotlinJsonMapper encoreToString with a MapAnySerializer then`() {
```

Comment:
```suggestion
    fun `Given the usage of any when invoking kotlinJsonMapper encoreToString with a MapAnySerializer then recommendation to avoid AnySerializer is logged`() {
```